WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignListView.swift

```diff
@@ -0,0 +1,140 @@
+import SwiftUI
+import struct Yosemite.Site
+
+/// Blaze campaign entry points.
+enum BlazeCampaignSource: String {
+    /// From the Blaze section on My Store tab.
+    case myStoreSection = "my_store_section"
+    /// From the Blaze campaign list
+    case campaignList = "campaign_list"
+}
+
+/// Hosting controller for `BlazeCampaignListView`
+///
+final class BlazeCampaignListHostingController: UIHostingController<BlazeCampaignListView> {
+
+    init(site: Site, viewModel: BlazeCampaignListViewModel) {
+        super.init(rootView: BlazeCampaignListView(viewModel: viewModel, siteURL: site.url.trimHTTPScheme()))
+
+        rootView.onCreateCampaign = { [weak self] in
+            let webViewModel = BlazeWebViewModel(source: .campaignList, site: site, productID: nil) {
+                viewModel.loadCampaigns()
+            }
+            let webViewController = AuthenticatedWebViewController(viewModel: webViewModel)
+            self?.navigationController?.show(webViewController, sender: self)
+        }
+    }
+
+    @available(*, unavailable)
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+/// View for showing a list of campaigns.
+///
+struct BlazeCampaignListView: View {
+    @ObservedObject private var viewModel: BlazeCampaignListViewModel
+    @State private var selectedCampaignURL: URL?
+
+    var onCreateCampaign: () -> Void = {}
+
+    private let siteURL: String
+
+    init(viewModel: BlazeCampaignListViewModel, siteURL: String) {
+        self.viewModel = viewModel
+        self.siteURL = siteURL
+    }
+
+    var body: some View {
+        Group {
+            switch viewModel.syncState {
+            case .results:
+                RefreshableInfiniteScrollList(spacing: Layout.contentSpacing,
+                                              isLoading: viewModel.shouldShowBottomActivityIndicator,
+                                              loadAction: viewModel.onLoadNextPageAction,
+                                              refreshAction: { completion in
+                    viewModel.onRefreshAction(completion: completion)
+                }) {
+                    ForEach(viewModel.campaigns) { item in
+                        BlazeCampaignItemView(campaign: item)
+                            .onTapGesture {
+                                let path = String(format: Constants.campaignDetailsURLFormat,
+                                                  item.campaignID, siteURL,
+                                                  BlazeCampaignSource.campaignList.rawValue)
+                                selectedCampaignURL = URL(string: path)
+                            }
+                    }
+                }
+            case .empty:
+                EmptyState(title: Localization.emptyStateTitle,
+                           description: Localization.emptyStateMessage,
+                           image: .emptyProductsImage)
```

Comment:
Nice work improvising empty state. 🎖️ I recommend asking design for a suitable empty blaze image as it will help promote Blaze better. (If you haven't already asked 😅)

Reply:
I pinged her again on Figma, will update this screen later in another PR if needed.